ubuntu-deploy/index.md

```diff
@@ -0,0 +1,414 @@
+
+## Installing Docker and Docker Compose
+
+To install Docker, run the following commands:
+
+```bash
+sudo apt install docker.io -y
+sudo systemctl start docker
+sudo systemctl enable docker
+```
+
+This installs Docker and enables the Docker service.
+To check if Docker is running, run:
+
+```bash
+sudo systemctl status docker
+```
+
+To install Docker Compose, run the following commands:
+
+```bash
+sudo apt install docker-compose -y
+```
+
+To verify that the installation was successful, you can run:
+
+```bash
+docker-compose --version
+```
+
+## Start your Directus Application
+
+Run these commands to allow incoming traffic on port 8055:
+
+```bash
+sudo ufw allow 8055/tcp
+sudo ufw enable
+```
+
+Start your Directus application using Docker Compose:
+
+```bash
+cd /path/to/your/directus/folder
+sudo docker-compose up
+```
+
+![Running docker compose][image-11]
+
+On the initial run, Docker will fetch the necessary image from the registry before launching your Directus application.
+
+Your application should now be accessible at `http://your_server_ip:8055`.
+
+![Directus application accessed with the server ip at port 8055][image-2]
+
+:::info Error Debugging
+
+If you encounter any error e.g `SQLITE_CANTOPEN: unable to open database file`, it is probably due to permission issues. You can try running this command to fix it:
+
+```bash
+sudo chown -R $USER:$USER /path/to/your/directus/folder
+
+```
+
+More info about this error in this [issue][issue]
+
+:::
+
+## Running the Docker Container as a Background Service
+
+Running your application like above is fine, but it will stop running when you close the terminal.
+
+To ensure your application runs in the background and restarts automatically, you can create a systemd service.
+
+:::info What is Systemd?
+
+Systemd is a system and service manager for Linux operating systems. It provides a standard process for controlling the startup, management, and monitoring of services and applications. It is usually defined by a configuration file usually ending with the _.service_ extension.
+
+:::
+
+### Create a Systemd Service File
+
+Create a file named _directus.service_ in the _/etc/systemd/system/_ directory:
+
+```bash
+sudo vi /etc/systemd/system/directus.service
+```
+
+Add the following content, updating the `WorkingDirectory` to your Directus project directory containing the _docker-compose.yml_ file:
+
+```plaintext
+[Unit]
+Description=Directus Docker Service
+Requires=docker.service
+After=docker.service
+
+
+[Service]
+Restart=always
+WorkingDirectory=/path/to/your/directory-containing-docker-compose
+ExecStart=/usr/bin/docker-compose up
+ExecStop=/usr/bin/docker-compose down
+
+
+[Install]
+WantedBy=multi-user.target
+```
+
+:::info
+
+You can get the full path to you directory by running the command `pwd` in the project directory on your server and copying the output.
+
+![print working directory(pwd) command][image-10]
```

Comment:
I don't think this image adds much value and would encourage you to remove it. Your description and the visual callout is more than enough.